cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/TimePartitionedFileSetDefinition.java

```diff
@@ -0,0 +1,109 @@
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.api.dataset.lib.AbstractDatasetDefinition;
+import co.cask.cdap.api.dataset.lib.CompositeDatasetAdmin;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.api.dataset.lib.FileSetArguments;
+import co.cask.cdap.api.dataset.lib.TimePartitionedFileSet;
+import co.cask.cdap.api.dataset.lib.TimePartitionedFileSetArguments;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.explore.client.ExploreFacade;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.inject.Inject;
+
+/**
+ * Defines the partitioned dataset type. At this time, the partitions are not managed by the
+ * partitioned dataset, so all admin is simply on the partition table.
+ */
+public class TimePartitionedFileSetDefinition extends AbstractDatasetDefinition<TimePartitionedFileSet, DatasetAdmin> {
+
+  private static final String PARTITION_TABLE_NAME = "partitions";
+  private static final String FILESET_NAME = "files";
+
+  private final DatasetDefinition<? extends Table, ?> tableDef;
+  private final DatasetDefinition<? extends FileSet, ?> filesetDef;
+
+  @Inject
+  private Injector injector;
+
+  public TimePartitionedFileSetDefinition(String name,
+                                          DatasetDefinition<? extends FileSet, ?> filesetDef,
+                                          DatasetDefinition<? extends Table, ?> tableDef) {
+    super(name);
+    Preconditions.checkArgument(tableDef != null, "Table definition is required");
+    Preconditions.checkArgument(filesetDef != null, "FileSet definition is required");
+    this.filesetDef = filesetDef;
+    this.tableDef = tableDef;
+  }
+
+  @Override
+  public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
+    return DatasetSpecification.builder(instanceName, getName())
+      .properties(properties.getProperties())
+      .datasets(filesetDef.configure(FILESET_NAME, properties),
+                tableDef.configure(PARTITION_TABLE_NAME, properties))
+      .build();
+  }
+
+  @Override
+  public DatasetAdmin getAdmin(DatasetSpecification spec, ClassLoader classLoader) throws IOException {
+    return new CompositeDatasetAdmin(filesetDef.getAdmin(spec.getSpecification(FILESET_NAME), classLoader),
+                                     tableDef.getAdmin(spec.getSpecification(PARTITION_TABLE_NAME), classLoader));
+  }
+
+  @Override
+  public TimePartitionedFileSet getDataset(DatasetSpecification spec,
+                                           Map<String, String> arguments, ClassLoader classLoader)
+    throws IOException {
+    // if the arguments do not contain an output location, generate one from the partition time
+    if (FileSetArguments.getOutputPath(arguments) == null) {
```

Comment:
maybe we should not allow getting the underlying fileset and instead add the required fileset methods to the time partitioned fileset api?